README
				 Gnome2::Rsvg
				--------------

This module allows you to use the Scalable Vector Graphics library (librsvg for
short) from Perl.

The index of the automatically generated API documentation can be accessed
with:

  perldoc Gnome2::Rsvg::index

				 Installation
				--------------

To install Gnome2::Rsvg run the following commands:

  perl Makefile.PL
  make
  make test
  make install

				 Dependencies
				--------------

Gnome2::Rsvg depends on:

  ExtUtils::Depends
  ExtUtils::PkgConfig
  Glib >= 1.02
  Gtk2 >= 1.02
  librsvg >= 2.12.0

If librsvg >= 2.14.0, we also depend on:

  Cairo >= 1.00


NEWS
Overview of changes in Gnome2::Rsvg 0.10
========================================

* Add support for rsvg_handle_render_cairo and rsvg_handle_render_cairo_sub.

Overview of changes in Gnome2::Rsvg 0.06
========================================

* Fix small build and test issues.

Overview of changes in Gnome2::Rsvg 0.05
========================================

* Track API changes in librsvg: Remove rsvg_pixbuf_from_file_*_ex and
  rsvg_handle_new_gz.  Require librsvg 2.12.0.
* Change some internal implementation details, update some documentation.

Overview of changes in Gnome2::Rsvg 0.04
========================================

* Fix various test issues.
* Bind new API: rsvg_handle_set_base_uri, rsvg_handle_get_base_uri, and
  rsvg_handle_get_metadata.

Overview of changes in Gnome2::Rsvg 0.03
========================================

* Bind and test rsvg_handle_get_title, rsvg_handle_get_desc,
  rsvg_set_default_dpi_x_y, and rsvg_handle_set_dpi_x_y.
* Make using gzip'ed images work by not using strlen() in
  Gnome2::Rsvg::Handle::write.

Overview of changes in Gnome2::Rsvg 0.02
========================================

* Miscellaneous small infrastructure and documentation improvements.
* Improved compatability with older versions of librsvg.


Rsvg.pm
# $Header: /cvsroot/gtk2-perl/gtk2-perl-xs/Gnome2-Rsvg/Rsvg.pm,v 1.9 2006/12/30 19:10:13 kaffeetisch Exp $

package Gnome2::Rsvg;

use 5.008;
use strict;
use warnings;

use Glib;
use Gtk2;

require DynaLoader;

our @ISA = qw(DynaLoader);
our $VERSION = '0.10';

sub dl_load_flags { $^O eq 'darwin' ? 0x00 : 0x01 }

Gnome2::Rsvg -> bootstrap($VERSION);

1;
__END__

=head1 NAME

Gnome2::Rsvg - Perl interface to the RSVG library

=head1 SYNOPSIS

  use Gnome2::Rsvg;

  my $svg = "path/to/image.svg";

  # The easy way.
  my $pixbuf = Gnome2::Rsvg -> pixbuf_from_file($svg);

  # The harder way.
  my $handle = Gnome2::Rsvg::Handle -> new();

  open(SVG, $svg) or die("Opening '$svg': $!");

  while (<SVG>) {
    $handle -> write($_) or die("Could not parse '$svg'");
  }

  close(SVG);

  $handle -> close() or die("Could not parse '$svg'");

  $pixbuf = $handle -> get_pixbuf();

=head1 ABSTRACT

This module allows a Perl developer to use the Scalable Vector Graphics library
(librsvg for short).

=head1 SEE ALSO

L<Gnome2::Rsvg::index>(3pm), L<Gtk2>(3pm), L<Gtk2::api>(3pm) and
L<http://librsvg.sourceforge.net/docs/html/index.html>

=head1 AUTHOR

Torsten Schoenfeld E<lt>kaffeetisch at gmx dot deE<gt>.

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2003-2005 by the gtk2-perl team

=cut


rsvg2perl.h
/*
 * Copyright (C) 2003-2005 by the gtk2-perl team
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 * $Header: /cvsroot/gtk2-perl/gtk2-perl-xs/Gnome2-Rsvg/rsvg2perl.h,v 1.4 2006/12/30 19:07:38 kaffeetisch Exp $
 */

#ifndef _RSVG2PERL_H_
#define _RSVG2PERL_H_

#include "rsvg2perl-version.h"

#include <gperl.h>
#include <gtk2perl.h>

#include <librsvg/rsvg.h>
#include <librsvg/librsvg-enum-types.h>

#if LIBRSVG_CHECK_VERSION (2, 14, 0)
# include <cairo-perl.h>
# include <librsvg/rsvg-cairo.h>
#endif

#include "rsvg2perl-autogen.h"

#endif /* _RSVG2PERL_H_ */


Makefile.PL
#!/usr/bin/perl -w
use 5.008;
use strict;
use ExtUtils::MakeMaker;
use Cwd;
use File::Spec;

# $Header: /cvsroot/gtk2-perl/gtk2-perl-xs/Gnome2-Rsvg/Makefile.PL,v 1.9 2006/12/30 19:07:38 kaffeetisch Exp $

our %build_reqs = (
  'perl-ExtUtils-Depends'   => '0.20',
  'perl-ExtUtils-PkgConfig' => '1.03',
  'perl-Glib'               => '1.02',
  'perl-Gtk2'               => '1.02',
  'perl-Cairo'              => '1.00', # if librsvg >= 2.14.0
  'librsvg'                 => '2.12.0',
);

our %pm_reqs = (
  'Gtk2'                => $build_reqs{'perl-Gtk2'},
  'Glib'                => $build_reqs{'perl-Glib'},
  'ExtUtils::Depends'   => $build_reqs{'perl-ExtUtils-Depends'},
  'ExtUtils::PkgConfig' => $build_reqs{'perl-ExtUtils-PkgConfig'},
);

unless (eval "use ExtUtils::Depends $build_reqs{'perl-ExtUtils-Depends'};"
           . "use ExtUtils::PkgConfig $build_reqs{'perl-ExtUtils-PkgConfig'};"
           . "use Glib '$build_reqs{'perl-Glib'}';"
           . "use Gtk2 '$build_reqs{'perl-Gtk2'}';"
           . "use Glib::MakeHelper;"
	   . "use Gtk2::CodeGen;"
           . "1") {
  warn "$@\n";
  WriteMakefile(
    PREREQ_FATAL => 1,
    PREREQ_PM    => \%pm_reqs,
  );
  exit 1; # not reached
}

my %rsvg_pkgcfg = ExtUtils::PkgConfig->find("librsvg-2.0 >= $build_reqs{librsvg}");

###############################################################################

my $need_cairo = 0;
my @rsvg_version = split /\./, $rsvg_pkgcfg{modversion};
if ($rsvg_version[0] > 2 || ($rsvg_version[0] == 2 && $rsvg_version[1] >= 14)) {
  $need_cairo = 1;
  $pm_reqs{Cairo} = $build_reqs{'perl-Cairo'};
  unless (eval "use Cairo '$build_reqs{'perl-Cairo'}'; 1;") {
    warn "$@\n";
    WriteMakefile(
      PREREQ_FATAL => 1,
      PREREQ_PM    => \%pm_reqs,
    );
    exit 1; # not reached
  }
}

###############################################################################

mkdir 'build', 0777;

our @xs_files = <xs/*.xs>;
our %pod_files = (
  'Rsvg.pm' => '$(INST_MAN3DIR)/Gnome2::Rsvg.$(MAN3EXT)',
  Glib::MakeHelper->do_pod_files (@xs_files),
);

Gtk2::CodeGen->parse_maps ('rsvg2perl');
Gtk2::CodeGen->write_boot (ignore => qr/^Gnome2::Rsvg$/);

ExtUtils::PkgConfig->write_version_macros (
  "build/rsvg2perl-version.h",
  "librsvg-2.0" => "LIBRSVG",
);

###############################################################################

my @deps = $need_cairo ? qw(Gtk2 Cairo) : qw(Gtk2);
my $rsvg = ExtUtils::Depends->new ('Gnome2::Rsvg', @deps);
$rsvg->set_inc ($rsvg_pkgcfg{cflags});
$rsvg->set_libs ($rsvg_pkgcfg{libs});
$rsvg->add_xs (@xs_files);
$rsvg->add_pm ('Rsvg.pm' => '$(INST_LIBDIR)/Rsvg.pm');
$rsvg->add_typemaps (map { File::Spec->catfile (cwd(), $_) }
                       'rsvg.typemap',
                       'build/rsvg2perl.typemap');

$rsvg->install (qw(rsvg2perl.h
                   build/rsvg2perl-autogen.h
                   build/rsvg2perl-version.h
                   doctypes));
$rsvg->save_config ('build/IFiles.pm');

WriteMakefile(
  NAME            => 'Gnome2::Rsvg',
  VERSION_FROM    => 'Rsvg.pm', # finds $VERSION
  ABSTRACT_FROM   => 'Rsvg.pm', # retrieve abstract from module
  XSPROTOARG      => '-noprototypes',
  MAN3PODS        => \%pod_files,
  PREREQ_PM       => \%pm_reqs,
  $rsvg->get_makefile_vars,
);

###############################################################################

sub MY::postamble {
  return Glib::MakeHelper->postamble_clean ()
       . Glib::MakeHelper->postamble_docs_full (
		DEPENDS => $rsvg,
		DOCTYPES => 'doctypes',
		COPYRIGHT_FROM => 'copyright.pod'
         )
       . Glib::MakeHelper->postamble_rpms (
		'GNOME_RSVG' => $build_reqs{'librsvg'},
		'PERL_EXTUTILS_DEPENDS' =>
			$build_reqs{'perl-ExtUtils-Depends'},
		'PERL_EXTUTILS_PKGCONFIG' =>
			$build_reqs{'perl-ExtUtils-PkgConfig'},
		'PERL_GLIB' => $build_reqs{'perl-Glib'},
		'PERL_GTK' => $build_reqs{'perl-Gtk2'},
	);
}
